src/censys/cloud_connectors/gcp_connector/connector.py

```diff
@@ -104,189 +107,402 @@ def scan_all(self):
             self.organization_id = provider_setting.organization_id
             self.scan()
 
-    def format_label(self, result: ListAssetsResponse.ListAssetsResult) -> str:
+    def list_projects(self) -> dict[str, dict]:
+        """List Gcp projects.
+
+        Returns:
+            dict[str, dict]: Gcp projects.
+        """
+        results = self.search_all_resources(filter=GcpCloudAssetInventoryTypes.PROJECT)
+        projects: dict[str, dict[str, str]] = {}
+        for result in results:
+            try:
+                project = ResourceSearchResult.to_dict(result)
+            except json.decoder.JSONDecodeError:  # pragma: no cover
+                self.logger.debug(f"Failed to parse project: {project}")
+                continue
+            try:
+                if versioned_resource := self.check_asset_version(
+                    GcpCloudAssetInventoryTypes.PROJECT, project
+                ):
+                    resource = versioned_resource["resource"]
+                    version = versioned_resource["version"]
+                    if version == "v3":
+                        project_id = resource["projectId"]
+                        project_number = self.parse_project_number(resource["name"])
+                        name = resource.get("displayName", "")  # Optional
+                    else:
+                        project_id = resource["projectId"]
+                        project_number = resource["projectNumber"]
+                        name = resource.get("name", "")  # Optional
+                    if (not project_id) or (not project_number):
```

Comment:
Another return above this if statement (hard to tell at a glance it's different than the top one).